src/simulators/statevector/statevector_state.hpp

```diff
@@ -551,19 +546,17 @@ void State<statevec_t>::set_config(const Config &config) {
 
   // Set threshold for truncating states to be saved
   json_chop_threshold_ = config.zero_threshold;
-  for (int_t i = 0; i < BaseState::qregs_.size(); i++) {
+  for (int_t i = 0; i < BaseState::qregs_.size(); i++)
     BaseState::qregs_[i].set_json_chop_threshold(json_chop_threshold_);
-  }
 
   // Set OMP threshold for state update functions
-  omp_qubit_threshold_ = config.statevector_parallel_threshold;
+  config.get_option(omp_qubit_threshold_, "statevector_parallel_threshold");
 
   // Set the sample measure indexing size
-  if (config.statevector_sample_measure_opt) {
-    int index_size = config.statevector_sample_measure_opt;
-    for (int_t i = 0; i < BaseState::qregs_.size(); i++) {
+  int index_size;
+  if (config.get_option(index_size, "statevector_parallel_threshold")) {
+    for (int_t i = 0; i < BaseState::qregs_.size(); i++)
```

Comment:
I don't think that this is a safe change - I think that previously if `index_size` was 0, the `if` wasn't entered, whereas now it's entered provided that `statevector_parallel_threshold` is set to anything, including zero.

Reply:
This is my mistake. It should be "statevector_sample_measure_opt" as a value of `int index_size`.